test/integration/test_contact_mechanics_biot.py

```diff
@@ -30,17 +30,84 @@ def _solve(self, setup):
 
     def test_pull_north_negative_scalar(self):
 
-        setup = SetupContactMechanicsBiot(
-            ux_south=0, uy_south=0, ux_north=0, uy_north=0.001
+        m = SetupContactMechanicsBiot()
+        m.uy_north = 0.001
+        m.with_fracture = False
+        m.mesh_args["mesh_size_bound"] = 0.5
+        u, p = self._solve(m)
+
+        # By symmetry (reasonable to expect from this grid), the average x displacement
+        # should be zero
+        self.assertTrue(np.abs(np.sum(u[0::2])) < m.zero_tol)
+        # Check that the expansion yields a negative pressure
+        self.assertTrue(np.all(p < -m.zero_tol))
+
+    def test_positive_reference_scalar(self):
+        """
+        Test nonzero (positive) p_reference. See GradP for documentation.
+        """
+        m = SetupContactMechanicsBiot()
+        m.with_fracture = False
+        m.nx = [4, 4]
+        m.fix_only_bottom = True
+        # Set nonzero reference state
+        m.p_reference = 1
+
+        u, p = self._solve(m)
+        # The initial and boundary conditions for pressure are 0. The absence of
+        # the positive reference pressure results in contraction of the medium,
+        # which is fixed at the bottom.
+        u_x = u[0::2]
+        u_y = u[1::2]
+        self.assertTrue(np.all(u_y < -m.zero_tol))
+        # Mirror indices left and right half of domain
```

Comment:
This will break at some point, e.g. with an update to `gmsh`. These are cell indices? So if we did the test on a Cartesian grid, symmetri is to be expected as well?

Reply:
I understand that the grids are Cartesian, so disregard this comment.